app/presenters/nav_tab/link_collection.rb

```diff
@@ -0,0 +1,123 @@
+class NavTab::LinkCollection
+  include Rails.application.routes.url_helpers
+  include TranslationHelper
+
+  attr_reader :ability, :facility
+
+  def initialize(facility, ability)
+    @facility = facility
+    @ability = ability
+  end
+
+  def admin
+    default + [
+      admin_orders,
+      admin_reservations,
+      admin_billing,
+      admin_products,
+      admin_users,
+      admin_reports,
+      admin_facility,
+    ].select(&:present?)
+  end
+
+  def customer
+    default + [orders, reservations, accounts]
+  end
+
+  def default
+    [home]
+  end
+
+  private
+
+  def accounts
+    NavTab::Link.new(tab: :accounts, text: t_my(Account), url: accounts_path)
+  end
+
+  def admin_billing
+    if !facility.all_facility? && ability.can?(:manage_billing, facility)
```

Comment:
Done, though I feel like we should do a bigger refactor related to all the `cross_facility` vs. `all_facility` changes. I'd be useful to have `#cross_facility?` and `#single_facility?`, and have `current_facility` set to a null-object `NilFacility` when outside a facility context.